examples/seismic/utils.py

```diff
@@ -140,15 +139,18 @@ def dtype(self):
 
     @property
     def rec(self):
-        return Receiver(name='rec', grid=self.grid,
+        return self.new_rec()
+
+    def new_rec(self, name='rec'):
+        return Receiver(name=name, grid=self.grid,
                         time_range=self.time_axis, npoint=self.nrec,
                         coordinates=self.rec_positions)
 
     @property
     def adj_src(self):
         if self.src_type is None:
-            warning("No surce type defined, returning uninitiallized (zero) shot record")
-            return self.rec
+            warning("No source type defined, returning uninitiallized (zero) shot record")
+            return self.new_rec()
```

Comment:
this looks off, because then this isn't a property, it's a method?
can we turn it into a cached_property or does mutability forbid this?

Reply:
Not sure. These are all method that just return new src/rec object based on the geometry, these will be read/written to in an operator so I think cached_property would make it quite unsafe .

Reply:
can we at least turn properties such as `adj_src` into actual methods?

Reply:
because conceptually they're not really _properties_ of a given Geometry instance